### PR TITLE
Fu a defragment - fixes issue #427

### PR DESCRIPTION
**Pull request by vidtec**
Resolves Issue #427 

Also adds some unit tests for NAL related items, and fully tested NAL reassembly.
Also introduces test coverage tool into build

### CODE REVIEW

File: src/test/java/org/jcodec/codecs/h264/io/model/NALReassembleTest.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+		
+		assertTrue("broken logic", out != null && out.length > 0);
+		assertArrayEquals("bad reassembly", out, result); 
+	}
+	
+	
+	@Test
+	public void testSimpleReassembleWorksStartAndEndOnly()
+	{
+		byte[][] data = new byte[][]
```
Comment (zhuker) [inline src/test/java/org/jcodec/codecs/h264/io/model/NALReassembleTest.java:156]: please use 4 spaces indent

File: src/test/java/org/jcodec/codecs/h264/io/model/NalUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import org.junit.Test;
+
+public class NalUnitTest 
+{
+
+    @Test
+    public void testCanReadNalUnitFromValidNal() 
+    {
+    	byte[] data = new byte[] { (byte)0x7C, (byte)0x81  };
```
Comment (zhuker) [inline src/test/java/org/jcodec/codecs/h264/io/model/NalUnitTest.java:15]: please use 4 spaces indentation

**Review by zhuker — CHANGES_REQUESTED** [review]
nice work!

please update formatting on unit tests, otherwise great!

**Comment by gareth-floodgate** [comment]
Whitespace formatting issues corrected in commit dd1fe9f - please go ahead and finalise for merge

Thanks

File: src/test/java/org/jcodec/codecs/h264/io/model/NalUnitTest.java (hunk 2 above)
Comment (zhuker) [reply src/test/java/org/jcodec/codecs/h264/io/model/NalUnitTest.java:15]: > Whitespace formatting issues corrected in commit [dd1fe9f](https://github.com/jcodec/jcodec/commit/dd1fe9fdad5346ee10f20114a331271efaaa2987) - please go ahead and finalise for merge
> 
> Thanks

NalUnitTest.java still uses 2 spaces @gareth-floodgate

**Comment by gareth-floodgate** [comment]
Apologies - updated this file in commit c6adfa3

**Review by zhuker — APPROVED** [review]

**Pull request merged**